app/policies/giveaway_policy.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class GiveawayPolicy < ApplicationPolicy
+  def show?
+    correct_user
+  end
+
+  def edit?
+    correct_user
+  end
+
+  def update?
+    correct_user
+  end
+
+  def destroy?
+    correct_user
+  end
+
+  private
+
+  def correct_user
+    @user == @record.user
```

Comment:
I'd suggest keeping conditions in action.

Reply:
does a before_action filter work in the ActionPolicy class?

Reply:
I'd suggest using `before_action` in the controller.

Reply:
Before_action filter doesn't work in the Policy Classes, if I want to use it in the controller I make another method with `authorize @giveaeay` and it gives wrong number of arguments. I can't seem to make it work with before action filter.

Reply:
> I'd suggest keeping conditions in action.

@violeta-p Should I repeat the code in every action, and delete the private method?

Reply:
Yes, that is my recommendation. In the future it would be easier to make changes.

Reply:
Ok